src/scitokens/utils/keycache.py

```diff
@@ -64,34 +64,40 @@ def getinstance():
             KEYCACHE_INSTANCE = KeyCache()
         return KEYCACHE_INSTANCE
 
-    def addkeyinfo(self, issuer, key_id, public_key, cache_timer=0):
+    def addkeyinfo(self, issuer, key_id, public_key, cache_timer=0, next_update=0):
         """
         Add a single, known public key to the cache.
         
         :param str issuer: URI of the issuer
         :param str key_id: Key Identifier
         :param public_key: Cryptography public_key object
         :param int cache_timer: Cache lifetime of the public_key
+        :param int next_update: Unix epoch timestamp for when the next update should occur
```

Comment:
Since the `cache_timer` is a lifetime relative to now - not a Unix timestamp - we should do the same with `next_update`.  Same throughout the rest of the changes.